python/cudf/cudf/core/multiindex.py

```diff
@@ -835,22 +835,9 @@ def size(self):
         return self._num_rows
 
     def take(self, indices):
-        if isinstance(indices, (Integral, Sequence)):
-            indices = np.array(indices)
-        elif isinstance(indices, cudf.Series) and indices.has_nulls:
-            raise ValueError("Column must have no nulls.")
```

Comment:
Removing this check will now result in a `RuntimeError`:

```python
>>> import cudf
>>> s = cudf.Series([1, 2, 3, 4, 5, 10])
>>> s.take([1, 1, None])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/frame.py", line 3038, in take
    return self._gather(indices, keep_index=keep_index)
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/frame.py", line 524, in _gather
    *libcudf.copying.gather(
  File "cudf/_lib/copying.pyx", line 175, in cudf._lib.copying.gather
RuntimeError: cuDF failure at: /nvme/0/pgali/cudf/cpp/src/copying/gather.cu:38: gather_map contains nulls
```

Should we retain this check?